3.24.x deps 2024 09 06 13 39 17 290986

craigcomstock opened the pull request:
- Updated libxml2 from 2.13.1 to 2.13.3
- Updated libcurl-hub from 8.8.0 to 8.9.1
- Updated apr from 1.7.4 to 1.7.5
- Updated apache from 2.4.59 to 2.4.62
- Updated git from 2.45.2 to 2.46.0
- Updated postgresql from 16.3 to 15.8
- Updated php from 8.3.8 to 8.3.11
- Updated libcurl from 8.8.0 to 8.9.1
- Updated libexpat from 2.5.0 to 2.6.3
- Updated openssl from 3.3.1 to 3.3.2

craigcomstock commented:
@cf-bottom jenkins no test

cf-bottom commented:
Sure, I triggered a build:

[![Build Status](https://ci.cfengine.com//buildStatus/icon?job=pr-pipeline&build=11186)](https://ci.cfengine.com//job/pr-pipeline/11186/)

 [NO TESTS]

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/11186/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-11186/

craigcomstock commented:
@cf-bottom jenkins no tests

cf-bottom commented:
Alright, I triggered a build:

[![Build Status](https://ci.cfengine.com//buildStatus/icon?job=pr-pipeline&build=11192)](https://ci.cfengine.com//job/pr-pipeline/11192/)

 [NO TESTS]

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/11192/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-11192/

craigcomstock commented:
@cf-bottom jenkins, think I'm ready for the tests

craigcomstock commented:
Oops. Forgot a fix somewhere from cf-bottom not quite working.
```
10:33:40 error: File /home/jenkins/workspace/testing-pr/label/PACKAGES_x86_64_linux_redhat_7/openssl/SOURCES/openssl-3.3.1.tar.gz: No such file or directory
```
will fix.

craigcomstock commented:
@cf-bottom jenkins

craigcomstock commented:
@cf-bottom jenkins

cf-bottom commented:
Sure, I triggered a build:

[![Build Status](https://ci.cfengine.com//buildStatus/icon?job=pr-pipeline&build=11202)](https://ci.cfengine.com//job/pr-pipeline/11202/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/11202/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-11202/

craigcomstock commented:
@vpodzime as you can see looking back in comments here we have a green jenkins build above with these changes.

I just squashed the two fixup commits for openssl into one commit with the upgrade.

Ready for review.

larsewi commented:
Maybe you can update the comment above

> Updated postgresql from 16.3 to 15.8

I got a bit confused at first ;)

vpodzime commented on build-scripts/install-dependencies:
```diff
@@ -40,13 +40,20 @@ check_and_install_perl()
+        PERL_OK="no"
+    fi
+    if ! perl -e 'use List::Util qw(pairs);'; then
+        echo "$PERL has List::Util that does not export pairs. Needs to be at least version 1.29 for OpenSSL version 3.3.2."
```
Hmm, AFAICT we check that Perl is >= 1.30 above. So this seems wrong.

craigcomstock replied:
This is List::Util version 1.29 or above.

The Perl version is not sufficient to check this particular aspect. We probably shouldn't have relied on a Perl version in the first place and instead put in exactly what was needed like this List::Util exporting pairs.

vpodzime reviewed (commented):
Looks good to me otherwise. Please check things work not just build on exotics (unless you already did).

craigcomstock commented:
> Maybe you can update the comment above
> 
> > Updated postgresql from 16.3 to 15.8
> 
> I got a bit confused at first ;)

Funny, that's the initial comment that github makes from the initial PR. That wrong version was generated by cf-bottom and then I fixed it. The commit comment is correct and now I have fixed the PR comment. :+1: Good catch.

craigcomstock commented:
> Looks good to me otherwise. Please check things work not just build on exotics (unless you already did).

Right. I will build and test on exotics.

craigcomstock commented:
check aix71 since it was offline
[![Build Status](https://ci.cfengine.com/buildStatus/icon?job=pr-pipeline&build=11214)](https://ci.cfengine.com/job/pr-pipeline/11214/)

olehermanse commented:
@craigcomstock why does PR title say 3.24.x but target branch is master?

craigcomstock commented:
> @craigcomstock please update the tables in the README (Version numbers in master column and add 3.24.x column).

Good catch. I used cf-bottom to start the PR but it did a bad job due to a few deps not working well for various reasons.

I will add a todo to the ticket to check this in all three branches: 3.21.x, 3.24.x and master.

craigcomstock commented:
> @craigcomstock why does PR title say 3.24.x but target branch is master?

No idea. I am not actively working on this PR right now, but rather the master one here: https://github.com/cfengine/buildscripts/pull/1493

So I will fix things up in due course.

olehermanse commented:
> I will add a todo to the ticket to check this in all three branches: 3.21.x, 3.24.x and master.

@craigcomstock I believe we've just updated the table on the master branch in the past (with correct versions from the other LTS branches).

craigcomstock commented:
reworked the commits from the master deps PR https://github.com/cfengine/buildscripts/pull/1493

craigcomstock commented:
Added cherry pick of 3.21.x removal of dependency table to this PR since that will go inline with this change as well.

@cf-bottom jenkins with exotics

cf-bottom commented:
Sure, I triggered a build:

[![Build Status](https://ci.cfengine.com//buildStatus/icon?job=pr-pipeline&build=11221)](https://ci.cfengine.com//job/pr-pipeline/11221/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/11221/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-11221/

craigcomstock commented:
retry hpux no-test job: [![Build Status](https://ci.cfengine.com/buildStatus/icon?job=pr-pipeline&build=11224)](https://ci.cfengine.com/job/pr-pipeline/11224/)

craigcomstock commented:
bootstrap-tests, restart failure is a flake on exotics where cf-serverd is not running, ENT-9767
sequential-tests two failures
- no-noise test, centos-7, ENT-12191
- cf-support on hpux, ENT-12192

vpodzime commented on deps-packaging/openssl/source:
```diff
@@ -1 +1 @@
-https://www.openssl.org/source/
+https://github.com/openssl/openssl/releases/download/openssl-3.3.2/
```
This change of the source is worth a comment in the commit message. Also, the commit message has a `cherry-picked from` reference to your private branch.

craigcomstock replied:
FWIW https://openssl-library.org/source/ links every release to github.com, there are no releases and openssl.org/source/ anymore.

vpodzime commented on build-scripts/install-dependencies:
```diff
@@ -78,12 +78,14 @@ check_and_install_perl()
-        $PATCH -p1 < $BASEDIR/buildscripts/build-scripts/perl-488307ffa6.patch
+        gzip -dc perl-${PERL_VERSION}.tar.gz | tar xf -
+        cd perl-${PERL_VERSION}
+#        $PATCH -p1 < $BASEDIR/buildscripts/build-scripts/perl-488307ffa6.patch
```
Time to drop this patching and the patch itself?

craigcomstock replied:
yeah, probably so. :+1:

vpodzime reviewed (commented):
I think the last commit should be part of the OpenSSL version bump. Also please remove the `cherry picked` references to private branches/commits.

craigcomstock commented:
> I think the last commit should be part of the OpenSSL version bump. Also please remove the `cherry picked` references to private branches/commits.

There are 4 commits related to OpenSSL version bump. Do you think all of them should be in one?

https://github.com/cfengine/buildscripts/pull/1494/commits/e3dc8b5beff7b6ada529b4183635596e76eca0db

https://github.com/cfengine/buildscripts/pull/1494/commits/d460ee01621cdd7abebb1c0e4da39ac38550491d

https://github.com/cfengine/buildscripts/pull/1494/commits/9910ebc4ad7aa7e19e2177e5ec88a647f65d224b

https://github.com/cfengine/buildscripts/pull/1494/commits/d4a1cb14f8c4bdb848b871edf7e6f145a31fee0e

craigcomstock commented:
This PR is proposed to merge to master. This is incorrect for this 3.24.x deps update. Will resubmit a new PR.
